### DISCUSSION TITLE
NeumannBC on a single node

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#25895] [Q&A Modules: Solid mechanics] [answered]
Hello,

I'd like to create a single node force in MOOSE. In order to apply a NeumannBC in MOOSE a sideset is needed. However I can't seem to figure out how to create a sideset consisting of only one node. How is that done?

**Comment by GiudGiud** [comment]
Hello

You cant create a sideset with a single node, you need an element side and in 2D that s 2 nodes at least.
You can create a nodeset, then look at using a NodalBC

Guillaume

**Reply by Flolaffel** [reply]
Is NodalBC the same thing as NodalKernel? If not where do I find NodalBCs?

Furthermore, is there a reason why you can apply NeumannBCs to only sidesets but can't make a single node sideset? Especially in regard of said use case of applying single node forces.

**Reply by GiudGiud** [reply]
It looks like it yes. i dont see nodalBCs as a system of its own, but nodalKernels can be boundary restricted. 

i think single-node forces are applied differently than NeumannBC but I dont remember how exactly. 
@hugary1995

**Reply by lindsayad** [reply]
A NeumannBC is an IntegratedBC. IntegratedBCs are naturally associated with surface terms in your weak form. NodalBCs perform strong constraint enforcement and are not associated with the weak form

**Reply by Flolaffel** [reply]
So how would I define a force NodalBC?

**Reply by maxnezdyur** [reply, marked as answer]
I think using a [`ConstantRate`](https://mooseframework.inl.gov/source/nodalkernels/ConstantRate.html) NodalKernel on that node would accomplish what you are trying to do.

**Reply by Flolaffel** [reply]
Works, thanks!

**Reply by Flolaffel** [reply]
What I'm wondering though: The documentation says `ConstantRate` implements a kernel du/dt. Wouldn't that be velocity instead of force? A test I've run shows it acts as a force though. Where is my error in reasoning here?

**Reply by maxnezdyur** [reply]
IIRC the `NodalKernel` adds directly to the residual of the of the nodes for your case that would be force. The example given there shows how to implement du/dt = -1. Where the time derivative  kernel computes du/dt and the ConstantRate add the -1.

**Reply by Flolaffel** [reply]
Ah I see. Thanks!
